File: golang/odd_number.go
package main

func counter(numbers []int) map[int]int {
	result := make(map[int]int)
	for i := 0; i < len(numbers); i++ {
		result[numbers[i]]++
	}

	return result
}

func FindOddNumber(numbers []int) int {
	counts := counter(numbers)
	nums := []int{}

	for k := range counts {
		nums = append(nums, k)
	}

	for i := 0; i < len(nums); i++ {
		if (counts[nums[i]])%2 != 0 {
			return nums[i]
		}
	}

	return 0
}

File: javascript/odd-number.js
/**
 * 
 * @param {number[]} numbers list of numbers example [7], [0], [1,1,2], [0,1,0,0,1], [1,2,3,4,5,6,7,8,9,0]
 * @returns {number} one number that odd number
 */

const counter = (numbers) => {
    const result = Object.create({})
    for (let i = 0; i < numbers.length; i++) {
        if (result[numbers[i]]) {
            result[numbers[i]]++
        } else {
            result[numbers[i]] = 1
        }
    }

    return result
}

const findOddNumber = (numbers) => {
    const counts = counter(numbers)
    const num = [...new Set(numbers)]

    for (let i = 0; i < Object.keys(counts).length; i++) {
        if (counts[num[i]] % 2 !== 0)
            return num[i]
    }

   return 0
}

module.exports = {
    findOddNumber
}